fuse/config.go

```diff
@@ -134,6 +138,8 @@ func PauseAndContract(confMap conf.ConfMap) (err error) {
 		return
 	}
 
+	globals.gate.Lock()
+
```

Comment:
there's a batch of "error out" cases below where we call return without dropping the lock. 

i'm assuming if we return an error we won't call ExpandAndResume() so perhaps its a moot point as to whether we release it now ... if we don't call ExpandAndResume() then the only other course of action is to exit ...

Hmmm.  the error cases in ExpandAndResume() probably have the same issue.  If we exit before we release the gate, who will release it?

Or does this not matter at all?

Reply:
Good point!   Probably it should be a defer Unlock()?